prefect_gcp/cloud_storage.py

```diff
@@ -452,14 +451,195 @@ def example_cloud_storage_copy_blob_flow():
         dest_blob = source_blob
 
     source_blob_obj = source_bucket_obj.blob(source_blob)
-    partial_copy_blob = partial(
+    await run_sync_in_worker_thread(
         source_bucket_obj.copy_blob,
         blob=source_blob_obj,
         destination_bucket=dest_bucket_obj,
         new_name=dest_blob,
         timeout=timeout,
         **copy_kwargs,
     )
-    await to_thread.run_sync(partial_copy_blob)
 
     return dest_blob
+
+
+class GcsBucket(ReadableFileSystem, WritableFileSystem):
+    """
+    Block used to store data using GCP Cloud Storage Buckets.
+
+    Attributes:
+        bucket: Name of the bucket.
+        gcp_credentials: The credentials to authenticate with GCP.
+        basepath: Used when you don't want to read/write at base level.
```

Comment:
This name mirrors the `GCS` block in the core library. We don't need to use this name, but I do think we want something more descriptive than `basepath` for this block.
```suggestion
        bucket_path: A default path within the GCS bucket to use for reading and writing objects.
```